extensions/amp-iframe/1.0/amp-iframe.js

```diff
@@ -18,34 +18,83 @@ class AmpIframe extends BaseElement {
     return super.isLayoutSupported(layout);
   }
 
+  /**
+   * Callback for onload event of iframe. Checks if the component has a placeholder
+   * element and hides it if it does. Also checks the position of the iframe on the
+   * document.
+   */
+  handleOnLoad_() {
+    if (this.getPlaceholder()) {
+      this.togglePlaceholder(false);
+      return;
+    }
+    // TODO(dmanek): Extract this to a common function & share
+    // between 0.1 and 1.0 versions.
+    measureIntersection(this.element).then((intersectionEntry) => {
+      const {top} = intersectionEntry.boundingClientRect;
+      const viewportHeight = intersectionEntry.rootBounds.height;
+      const minTop = Math.min(600, viewportHeight * 0.75);
+      userAssert(
+        top >= minTop,
+        '<amp-iframe> elements must be positioned outside the first 75% ' +
+          'of the viewport or 600px from the top (whichever is smaller): %s ' +
+          ' Current position %s. Min: %s' +
+          "Positioning rules don't apply for iframes that use `placeholder`." +
+          'See https://github.com/ampproject/amphtml/blob/main/extensions/' +
+          'amp-iframe/amp-iframe.md#iframe-with-placeholder for details.',
+        this.element,
+        top,
+        minTop
+      );
+    });
+  }
+
+  /**
+   * Updates the element's dimensions to accommodate the iframe's
+   * requested dimensions.
+   * @param {number} height
+   * @param {number} width
+   * @private
+   */
+  updateSize_(height, width) {
+    if (!height && !width) {
+      this.user().error(
```

Comment:
This error won't be called because of the early return in the Preact component based on the same conditions. Should we move this message to a `console.error` call in that layer instead? https://github.com/ampproject/amphtml/blob/83570bb73262f5da5fd1f5f034b463c8b325c73a/extensions/amp-iframe/1.0/component.js#L36-L38

Reply:
Good catch. Added `console.error` in the preact layer and removed this check from here.